abstract/group.go

```diff
@@ -49,6 +49,9 @@ type Scalar interface {
 
 	// Set to the modular inverse of scalar a
 	Inv(a Scalar) Scalar
+	
+	// Set to the gcd of a and its module
+	Gcd(a Scalar) Scalar
```

Comment:
The main reasons the whole interface of `Scalar` is defined like this is to avoid un-necessary allocations. You can for example use a `var tmp = suite.Scalar()` which holds  different steps of a computations without changing or cloning important scalars such as a private key etc. This reasoning is from `big.Int` library, and applies here as well. So for a GCD, it should be the same,i.e. it should be possible to provide both parameters in the arguments and let the caller decides which scalar will be the recipient.
I'm ok with keeping that in the `Scalar` interface also.
For the x and y then it's fine for me also :)